### ISSUE TITLE
Make malware_types optional

### ISSUE THREAD

**Issue by johnwunder** [oasis-tcs/cti-stix2#211]
The MITRE ATT&CK infrastructure and content team has been looking at how they can upgrade to STIX 2.1 when the time comes. Everything seems very straightforward, except for one thing: `malware_types` is now required.

ATT&CK uses `malware` as one of the two objects (`tool` being the other) to categorize our Software pages: https://attack.mitre.org/software/.

Right now, we don't provide anything like malware types. Our main purpose is to categorize malware against ATT&CK techniques, of course, and so having to go through reporting to also categorize the types is extra work. We also worry that it won't be super accurate, since this isn't primary source reporting and we might not have all the reporting....so it would be easy to miss things. Many of the categories are also duplicative with ATT&CK techniques, and so we'd be representing the information twice and it may become duplicative.

If this change isn't adopted, the approach will be to provide a value of `unknown` in the list...which is fine. But unknown is essentially optional and so it might be cleaner to just make the field optional in case other producers have the same challenges.

**Comment by jordan2175** [comment]
Let's talk about this on the next working call.

**Comment by allant0** [comment]
i'm not sure what is actually the proposed change? 

I appreciate that MITRE have to do extra work but thats actually true of any organization (Cisco, Fireeye, Symantec...etc) that has already significant investment in non-STIX-based intelligence content that include meta-data descriptions and intelligence context that is deeper than a simple IP or Domain technical indicator.  Mapping between our verbose taxonomies and meta-data to STIX2 data model is a cost that just had to be done. In some cases, there was not a direct 1-1 mapping either. 

Are you suggesting malware_types is made optional? If yes, then we would need clear description on what it means when that content is missing in malware. 

If you are suggesting changing to a different mapping then need to understand what the new definitions would look like.

**Comment by jordan2175** [comment]
We talked about this on the 2020-01-28 working call.  The consensus was to make this optional but leave the value of unknown in the vocab.  We will need to add text to make sure it is clear what that means and that you can only use unknown and nothing else at the same time, or something.  We need to look at other vocabs and other uses and see where else this is a problem.  The consensus on the call was to look at other places as well and keep them consistent.  Like tool indicator.

**Comment by jordan2175** [comment]
Per the working call, we made changes to the following objects: Indicator (indicator_types), Infrastructure (infrastructure_type), Malware (malware_types), Report (report_types), Threat Actor (threat_actor_types), Tool (tool_types), Identity (identity_class). We did NOT change the Relationship (relationship_type) as that one has a slightly different meaning and is actually required to make sense of the relationship. 

We changed the following vocabulary entries to match the discussion on the working call.  Infrastructure Vocab Undefined -> Unknown
Identity Vocab - Unspecified -> Unknown and then updated the description.
